### DISCUSSION TITLE
Feature Request: Range selection

### DISCUSSION THREAD

**Discussion by ksnyder** [JedWatson/react-select#4428] [Ideas]
I love this library and it would be great if users could select a range of options (by holding down Shift or some other key) instead of having to build up their selections one at a time.  The use case is a long list of options where users always select a large number of options (about 10-40), and those options are usually in contiguous blocks.  For options not in a contiguous range, they could se the Ctrl key to select individual options.  Thanks.

**Comment by cristi0212** [comment]
It would be good if range will be implemented, like on react-super-select.
With Shift + click to be selected entire range.
Tkank you!

**Comment by ebonow** [comment]
Moved from issues to discussions

**Comment by arthurmmedeiros** [comment]
Is there any update on this subject?

**Reply by ebonow** [reply]
This isn't currently a priority item on the roadmap as there are other stability issues being addressed by a hand full of volunteer maintainers.

Additionally, this seems like something that can be done without any changes to the underlying library so adding a new prop to shortcut the work for this likely isn't something that would be accepted. Instead there is roadmap for creating Select hooks that can add composable behaviors and components and this would likely fall in that category. 

However, if there is more traction on this regarding upvotes, it would likely be considered sooner than later.

**Reply by Bobfrat** [reply]
Are there examples of how to implement this using the current library?
